api/src/test/java/demo/BaseIntegrationTest.java

```diff
@@ -43,9 +48,18 @@ public abstract class BaseIntegrationTest {
     @Autowired
     protected UserService userService;
 
+    @Autowired
+    protected FilterService filterService;
```

Comment:
My idea is that, depending on the TEST scenario, the code will call and create test entities (users, filters,... etc). And since they are very boilerplate/ duplicated code, then combining it into 1 place makes sense. If a test breaks due to this, then its likely a edge case and that test can have it´s own test data

But I do agree that one test should not depend on another test, whether it succeeds or not.